File: app/src/main/java/org/doomer/qnotez/adapters/RecyclerViewAdapter.java
package org.doomer.qnotez.adapters;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.doomer.qnotez.App;
import org.doomer.qnotez.R;
import org.doomer.qnotez.db.NoteModel;
import org.doomer.qnotez.utils.TextUtils;

import java.text.SimpleDateFormat;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewHolder> {

    private List<NoteModel> noteModelList;
    private View.OnClickListener clickListener;
    private View.OnLongClickListener longClickListener;


    static class RecyclerViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.item_title)
        protected TextView txtTitle;

        @BindView(R.id.item_text)
        protected TextView txtText;

        @BindView(R.id.item_updated)
        protected TextView txtUpdated;

        public RecyclerViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }


    public RecyclerViewAdapter(List<NoteModel> list,
                               View.OnClickListener listener,
                               View.OnLongClickListener longListener) {
        noteModelList = list;
        clickListener = listener;
        longClickListener = longListener;
    }

    @Override
    public RecyclerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new RecyclerViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.recycler_item, parent, false));
    }

    @Override
    public void onBindViewHolder(final RecyclerViewHolder holder, int position) {
        // get item from list by positin
        NoteModel note = noteModelList.get(position);

        int rowCount = countOfDisplayedTextRows();
        switch (rowCount) {
            case 0:
                holder.txtText.setVisibility(View.INVISIBLE);
                break;
            default:
                break;
        }

        // bind values to view in holder
        if (rowCount != 0) {
            String text = TextUtils.rowsToPreview(rowCount, note.getText());
            holder.txtText.setText(text);
        }

        String title = note.getTitle();

        if (title.isEmpty()) {
            title = holder.txtTitle.getContext().getString(R.string.iten_no_title);
        }

        holder.txtTitle.setText(title);

        SimpleDateFormat sdf = new SimpleDateFormat("dd MMM yyyy k:mm");
        String createdString = sdf.format(note.getUpdated());

        holder.txtUpdated.setText(createdString);

        holder.itemView.setTag(note);
        holder.itemView.setOnClickListener(clickListener);
        holder.itemView.setOnLongClickListener(longClickListener);
    }

    private int countOfDisplayedTextRows() {
        Context ctx = App.getAppContext();

        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(ctx);
        String key = ctx.getString(R.string.settings_key_count_rows);

        return Integer.parseInt(sp.getString(key, "-1"));
    }

    @Override
    public int getItemCount() {
        return noteModelList.size();
    }

    public void addItems(List<NoteModel> noteList) {
        noteModelList = noteList;
        notifyDataSetChanged();
    }


}

File: app/src/main/kotlin/org/doomer/qnotez/adapters/RecyclerViewAdapter.kt
package org.doomer.qnotez.adapters

import android.content.SharedPreferences
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import org.doomer.qnotez.di.App
import org.doomer.qnotez.R
import org.doomer.qnotez.db.NoteModel
import org.doomer.qnotez.utils.TextUtils

import java.text.SimpleDateFormat

import kotlinx.android.synthetic.main.recycler_item.view.*


class RecyclerViewAdapter(private var noteModelList: List<NoteModel>,
                          private var sp: SharedPreferences,
                          private val clickListener: View.OnClickListener,
                          private val longClickListener: View.OnLongClickListener)
    : RecyclerView.Adapter<RecyclerViewAdapter.RecyclerViewHolder>() {

    class RecyclerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var item_title = itemView.item_title
        var item_text = itemView.item_text
        var item_updated = itemView.item_updated
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        return RecyclerViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.recycler_item, parent, false))
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
        // get item from list by positin
        val note = noteModelList!![position]

        val rowCount = countOfDisplayedTextRows()
        when (rowCount) {
            0 -> holder.item_text.visibility = View.INVISIBLE
            else -> {
            }
        }

        // bind values to view in holder
        if (rowCount != 0) {
            val text = TextUtils.rowsToPreview(rowCount, note.text!!)
            holder.item_text.text = text
        }

        var title = note.title

        if (title!!.isEmpty()) {
            title = holder.item_text.context.getString(R.string.iten_no_title)
        }

        holder.item_title.text = title

        val sdf = SimpleDateFormat("dd MMM yyyy k:mm")
        val createdString = sdf.format(note.updated)

        holder.item_updated.text = createdString

        holder.itemView.tag = note
        holder.itemView.setOnClickListener(clickListener)
        holder.itemView.setOnLongClickListener(longClickListener)
    }

    private fun countOfDisplayedTextRows(): Int {

        // TODO - make get key from string resvia app context
        val key = App.appContext!!.getString(R.string.settings_key_count_rows)

        return Integer.parseInt(sp.getString(key, "-1"))
    }

    override fun getItemCount(): Int {
        return noteModelList!!.size
    }

    fun addItems(noteList: List<NoteModel>) {
        noteModelList = noteList
        notifyDataSetChanged()
    }


}
